build.sbt

```diff
@@ -37,23 +37,21 @@ resolvers ++= Resolver.sonatypeOssRepos("releases")
 libraryDependencies ++= Seq(
   "com.gu" %% "content-api-models-scala" % "25.0.0",
   "com.gu" %% "thrift-serializer" % "5.0.5",
-  "software.amazon.kinesis" % "amazon-kinesis-client" % "2.6.0",
+  "software.amazon.kinesis" % "amazon-kinesis-client" % "3.0.1",
   "com.typesafe.scala-logging" %% "scala-logging" % "3.9.5",
   "com.twitter" %% "scrooge-core" % "21.12.0",
   "org.scalatest" %% "scalatest" % "3.2.19" % Test
-)
+) ++ Seq("aws-json-protocol", "dynamodb", "kinesis").map(artifact => "software.amazon.awssdk" % artifact % "2.29.23")
 
 val jacksonVersion = "2.17.2"
 dependencyOverrides ++= Seq(
   "com.charleskorn.kaml" % "kaml" % "0.53.0",
   "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion,
   "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVersion,
   "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
-  "software.amazon.awssdk" % "netty-nio-client" % "2.26.25",
   "org.json" % "json" % "20231013",
   "org.xerial.snappy" % "snappy-java" % "1.1.10.4",
   "org.apache.commons" % "commons-compress" % "1.26.0",
-  "com.amazon.ion" % "ion-java" % "1.10.5",//overriding until a version of amazon-kinesis-client is available that removes the ion-java vulnerability
```

Comment:
`ion-java` had a dependency override added with #42 in March 2024. `ion-java` was a dependency of AWS SDK v1 (`com.amazonaws:aws-java-sdk-core:1.12.151`). However, release [3.0.0](https://github.com/awslabs/amazon-kinesis-client/releases/tag/v3.0.0) of the Amazon Kinesis Client Library for Java removed the last transitory dependency of the KCL on AWS SDK v1, and so `content-api-firehose-client` no longer depends upon it.